programs/marginfi/src/state/marginfi_account.rs

```diff
@@ -101,6 +101,17 @@ impl MarginfiAccount {
         );
         Ok(())
     }
+
+    pub fn can_be_closed(&self) -> bool {
+        let is_disabled = self.get_flag(DISABLED_FLAG);
```

Comment:
why must it be not disabled?

Reply:
oh, because they can close and reopen...

Reply:
Are we recording insurance fund shares in the users `MarginfiAccount`? My understanding of the feedback was not to include the `MarginfiAccount` as part of the ixs and so it was removed. 

However, if we are not sending tokenized shares, we would need some kind of accounting system to keep track of who has how many shares I believe. Maybe doing so in the liquid insurance fund makes most sense?

Reply:
I would circle back with jakob on this front. Here are my thoughts:

1) Under tokenization it makes sense to not record shares because that's what the token program is doing. As you mentioned, you need accounting for this if not tokenizing.
2) You never want unbounded account structures. If you record this in the fund, it will be a perpetually shrinking/growing data type.

With this, it seems we need a per-user account for the accounting. Right now the insurance fund ux is fairly separated from other user actions in mrgnlend, so on one hand it may make sense for a new account type to minimize attack surface area. On the other hand, if in the future these insurance fund actions are more tied together to other user actions, it will only add another account to our already-large number of accounts for some instructions.

Those are the immediate for/against I see for separate user accounts.

Reply:
Yes, agreed on all points. We landed on a new PDA account per user that uses seeds from the user pk and the bank. This seems like a good stopgap solution for now. Upon review we can see if we want to adjust things. 

Personally I think using the existing MarginfiAccount type would be easier to integrate into the UI, but let's see.

Reply:
Shares shouldn't be on the marginfi account anyway so closing account is not applicable.

Reply:
Shares shouldn't be on the marginfi account anyway so closing account is not applicable.

Reply:
Resolving this as its not relevant